test/unit-test/ota_utest.c

```diff
@@ -2048,6 +2125,33 @@ void test_OTA_SelfTestJobDowngrade()
     TEST_ASSERT_EQUAL( true, resetCalled );
 }
 
+void test_OTA_SelfTestJobSameVersion()
+{
+    pOtaJobDoc = JOB_DOC_SELF_TEST_SAME_VERSION;
+
+    otaGoToState( OtaAgentStateWaitingForJob );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
+
+    otaReceiveJobDocument();
+    receiveAndProcessOtaEvent();
+    TEST_ASSERT_EQUAL( true, resetCalled );
+}
+
+void test_OTA_SelfTestJobFileType()
```

Comment:
Might want to rename to reflect the condition tested?  `test_OTA_SelfTestJobNonFirmwareFileType`? or add a comment with the scenario.

Reply:
done